.github/workflows/ci.yaml

```diff
@@ -11,21 +11,42 @@ env:
   rust_toolchain: nightly-2022-02-09
 
 jobs:
-  build-rust:
-    name: "Rust Build"
+  format-rust:
+    name: "Rust Formatting"
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout code
+        uses: actions/checkout@v2
       - name: Install supported Rust nightly
         uses: actions-rs/toolchain@v1
         with:
           toolchain: ${{ env.rust_toolchain }}
           profile: minimal
           override: true
-          components: rustfmt, rust-src
-      - uses: Swatinem/rust-cache@v1
+          components: rustfmt, clippy
+      - name: Cache Rust package builds
+        uses: Swatinem/rust-cache@v1
       - name: Check Rust formatting
         run: cargo fmt -- --check
+      - name: Check Clippy linting
+        run: cargo clippy --all --tests --all-features -- -A clippy::needless_option_as_deref
```

Comment:
Can fix the `clippy::needless_option_as_deref` issue with `PyMessage` by upgrading PyO3 to [v0.16](https://pyo3.rs/master/changelog.html#fixed-2). I added a task to address this in a future PR.